clients/src/main/java/org/apache/kafka/common/protocol/ProtoUtils.java

```diff
@@ -27,7 +27,7 @@ private static Schema schemaFor(Schema[][] schemas, int apiKey, int version) {
         if (apiKey < 0 || apiKey > schemas.length)
             throw new IllegalArgumentException("Invalid api key: " + apiKey);
         Schema[] versions = schemas[apiKey];
-        if (version < 0 || version > versions.length)
+        if (version < 0 || version >= versions.length)
             throw new IllegalArgumentException("Invalid version for API key " + apiKey + ": " + version);
         if (versions[version] == null)
```

Comment:
I believe `versions[version]` will result in an out of bounds exception if `version` is greater than `versions.length`.